cmd/tempo/app/config.go

```diff
@@ -76,6 +77,7 @@ func (c *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet) {
 	f.BoolVar(&c.UseOTelTracer, "use-otel-tracer", false, "Set to true to replace the OpenTracing tracer with the OpenTelemetry tracer")
 	f.BoolVar(&c.EnableGoRuntimeMetrics, "enable-go-runtime-metrics", false, "Set to true to enable all Go runtime metrics")
 	f.BoolVar(&c.AutocompleteFilteringEnabled, "autocomplete-filtering.enabled", true, "Set to false to disable autocomplete filtering")
+	f.DurationVar(&c.ShutdownDelay, "shutdown-delay", 0, "How long to wait between SIGTERM and shutdown. After receiving SIGTERM, Tempo will report not-ready status via /ready endpoint.")
```

Comment:
Q: do we want to set a non zero default here?? so folks have a clean shutdown by default?

Reply:
i prefer to 0 for a few reasons:
- consistency: mimir defaults to 0
- just setting this param alone doesn't guarantee a clean shutdown.  it requires coordination with your orchestration platform
- a default of 0 means there's no change in behavior

Reply:
makes sense.